client/my-sites/themes/controller.js

```diff
@@ -68,6 +69,16 @@ function makeElement( ThemesComponent, Head, store, props ) {
 	);
 };
 
+// Where do we put this? It's client/server-agnostic, so not in client/controller,
+// which requires ReactDom... Maybe in lib/react-helpers?
+const makeLayout = ( store, primary, secondary, tertiary ) => (
```

Comment:
Maybe we'll turn this into actual middleware for explicit invocation in `my-sites/themes/index.js`, or even call it implicitly from `client/controller#renderLayout`. We could then have individual middleware gradually populating `context.primary`, `secondary`, and `tertiary` similar to what we have now, i.e. e.g.

```js
'/design/type/:tier': [ multiSite, makeNavigation, makeSiteSelection, makeLayout ]
```

where `makeNavigation` and `makeSiteSelection` put their element trees into `context.secondary`.